### DISCUSSION TITLE
TestClient Issue with client.delete()

### DISCUSSION THREAD

**Discussion by FellowFellow** [fastapi/fastapi#10689] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, Form
from fastapi.responses import JSONResponse
from fastapi.testclient import TestClient


app = FastAPI()

@app.delete("/test_delete")
async def test_delete(
    some_data: str = Form(...)
) -> JSONResponse:
    return JSONResponse({"some_data": some_data})


test_client = TestClient(app)

test_client.delete("/test_delete", data={'some_data': "Test Data"})
```

### Description

# Bug Case

## Steps to reproduce
1. install packages `pip install fastapi httpx`
2. run file.py

output:
```
    test_client.delete("/test_delete", data={'some_data': "Test Data"})
TypeError: TestClient.delete() got an unexpected keyword argument 'data'
```

## My system
MacOs & Ubuntu 
tested python versions: `3.8.6`, `3.9.16`, `3.11.4`, 3.12.0` 

### Operating System

Linux, macOS

### Operating System Details

tested python versions: `3.8.6`, `3.9.16`, `3.11.4`, 3.12.0` 

### FastAPI Version

0.104.1

### Pydantic Version

2.5.0

### Python Version

`3.8.6`, `3.9.16`, `3.11.4`, 3.12.0`

**Comment by Zer0x00** [comment, marked as answer]
Please see https://www.python-httpx.org/compatibility/#request-body-on-http-methods and https://github.com/encode/httpx/discussions/1587#discussioncomment-639160
